usr/lib/cca_stdll/cca_specific.c

```diff
@@ -235,33 +236,35 @@ CK_RV token_specific_rng(STDLL_TokData_t * tokdata, CK_BYTE * output,
                          CK_ULONG bytes)
 {
     long return_code, reason_code;
-    unsigned char form[CCA_KEYWORD_SIZE], random_number[CCA_RNG_SIZE];
-    CK_ULONG bytes_so_far = 0, bytes_left;
+    unsigned char rule_array[CCA_KEYWORD_SIZE];
+    CK_ULONG bytes_so_far = 0, num_bytes;
+    long rule_array_count = 1, zero = 0;
     CK_RV rv;
 
     UNUSED(tokdata);
 
-    memcpy(form, "RANDOM  ", (size_t) CCA_KEYWORD_SIZE);
+    memcpy(rule_array, "RANDOM  ", (size_t) CCA_KEYWORD_SIZE);
 
     while (bytes_so_far < bytes) {
-        dll_CSNBRNG(&return_code,
-                    &reason_code, NULL, NULL, form, random_number);
+        num_bytes = bytes - bytes_so_far;
```

Comment:
loop condition feat. 3 vars (num_bytes, bytes, bytes_so_far) hard to read. why not
```
while (bytes) {
    num_bytes = bytes > 8192 ? 8192 : bytes;
    [....]
    bytes -= num_bytes;
}
```

Reply:
Well, there are many ways to do this...
I guess I just wanted to keep it the way it was before as much as possible.